### ISSUE TITLE
[Feature Req.] Implement IteratorAggregate in Collection

### ISSUE THREAD

**Issue by stemis** [stripe/stripe-php#575]
Hello,

I am writing this issue to suggest the use of `IteratorAggregate` to the `Collection` class.
I would like to be able to use the `Collection` as follows:
```php
foreach($customer->sources as $source) {
...
}
```

Since `Collection` is not implementing a `Traversable`, this is currently not possible.

Right now iteration is possible using the following methods:
```php
foreach ($customer->sources->data as $card) {
    ...
 }

// or

foreach ($customer->sources->autoPagingIterator() as $source) {
    ...
 }
```

Since `$collection->autoPagingIterator()` is already returning an iterator, it seems like a small request to add the `getIterator()` required by the `IteratorAggregate` interface to the `Collection` class.

Is this something that seems feasible?

Kind Regards,
Steff

**Comment by ob-stripe** [comment]
Sounds very reasonable! In fact, I thought you could already iterate directly over `Collection` instances 😄 We'll implement this soon.

**Comment by stemis** [comment]
Thanks! Much appreciated 😄

**Comment by ob-stripe** [comment]
Implemented in 6.28.0.
